pkg/report/sarif.go

```diff
@@ -259,6 +286,20 @@ func ToPathUri(input string) string {
 	return strings.ReplaceAll(input, "\\", "/")
 }
 
+func getLocations(name, version string, pkgs []ftypes.Package) []location {
```

Comment:
`getLocations` calls for every vulnerability, so if a package contains several vulns these `for`-loops will run several times for the same data. right? or i missed something. 

as a solution, we can create a map with `package-version` key and `locations` value.
Wdyt?

Reply:
good idea.
I used `<path>@<name>@<version>` for map.
It currently looks better. Take a look, please.